keps/1136-provisioning-request-support/README.md

```diff
@@ -0,0 +1,185 @@
+# KEP-1136: ProvisioningRequest support
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Notes/Constraints/Caveats (Optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+Introduce an (AdmissionCheck)[https://github.com/kubernetes-sigs/kueue/tree/main/keps/993-two-phase-admission]
+that will use (`ProvisioningRequest`)[https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/proposals/provisioning-request.md]
+to ensure that there is enough capacity in the cluster before 
+admitting a workload.
+
+## Motivation
+
+Currently Kueue admits workloads based on the quota check alone. 
+This works reasonably well in most cases, but doesn't provide 
+guarantee that an admitted workload will actually schedule 
+in full in the cluster. With `ProvisioningRequest`, SIG-Autoscaling owned
+(ClusterAutoscaler)[https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler]
+opens a way for stronger (but still not hard-guaranteed) all-or-nothing
+scheduling in an autoscaled cloud environment. 
+
+Before admission, CA will check whether there is enough resources and
+provide them if their number is not sufficient (details
+depend on the exact engine used with `ProvisioningRequest)`.
+
+### Goals
+
+* Provide Kueue integration with `ProvisioningReqest`.
+* Define how users can configure what Kueue puts into `ProvisioningRequest`.
+* Provide a list of test cases to ensure that all corner cases are covered.
+
+### Non-Goals
+
+* Define how Cluster Autoscaler handles ProvisioningRequest.
+* Define underling cloud-specific behavior.
+
+## Proposal
+
+* Introduce a new controller in Kueue that will act as AdmissionCheck based on
+  the status of created `ProvisioningRequest`.
+
+* Introduce a new cluster-scoped CRD to configure how `ProvisioningRequest` should be used.
```

Comment:
global or per ClusterQueue?

Reply:
it's referenced in the `AdmissionCheck` using it.